File: frontend/src/service/match/matchResultService.ts
import api from "../../config/axios.ts";
import type { ApiResponse } from "../../types/ApiResponse.ts";
import type {
  MatchResultRequest,
  MatchResultResponse,
} from "../../types/matchResult.ts";

const API_BASE_URL = "/match-results";

export const matchResultService = {
  submitMatchResult: async (
    matchId: string,
    winningTeamNumber: number,
    absentUserIds: string[] = [],
  ): Promise<ApiResponse<MatchResultResponse>> => {
    const payload: MatchResultRequest = {
      matchId,
      winningTeamNumber,
      absentUserIds,
    };
    const response = await api.post(`${API_BASE_URL}/submit`, payload);
    return response.data;
  },

  respondToResult: async (
    resultId: string,
    isAccepted: boolean,
  ): Promise<ApiResponse<MatchResultResponse>> => {
    const response = await api.post(
      `${API_BASE_URL}/${resultId}/respond`,
      null,
      {
        params: { isAccepted },
      },
    );
    return response.data;
  },

  getResultsByMatch: async (
    matchId: string,
  ): Promise<ApiResponse<MatchResultResponse[]>> => {
    const response = await api.get(`${API_BASE_URL}/match/${matchId}`);
    return response.data;
  },
};

export default matchResultService;

File: mobile/lib/services/match_result_service.dart
import '../config/api_client.dart';
import '../models/match_result.dart';

class MatchResultService {
  final String _baseUrl = '/match-results';

  Future<MatchResultResponse> submitMatchResult(
      String matchId,
      int winningTeamNumber, [
        List<String> absentUserIds = const [],
      ]) async {
    final payload = MatchResultRequest(
      matchId: matchId,
      winningTeamNumber: winningTeamNumber,
      absentUserIds: absentUserIds,
    );

    final res = await apiClient.post('$_baseUrl/submit', data: payload.toJson());
    return MatchResultResponse.fromJson(res.data['result']);
  }

  Future<MatchResultResponse> respondToResult(String resultId, bool isAccepted) async {
    final res = await apiClient.post(
      '$_baseUrl/$resultId/respond',
      queryParameters: {'isAccepted': isAccepted},
    );
    return MatchResultResponse.fromJson(res.data['result']);
  }

  Future<List<MatchResultResponse>> getResultsByMatch(String matchId) async {
    final res = await apiClient.get('$_baseUrl/match/$matchId');
    final data = res.data['result'] as List;
    return data.map((json) => MatchResultResponse.fromJson(json)).toList();
  }
}

final matchResultService = MatchResultService();